Content.Client/_DV/Curation/UI/Cwoink/CwoinkControl.xaml.cs

```diff
@@ -0,0 +1,282 @@
+            var bRecent = b.ActiveThisRound && bch.LastMessage != DateTime.MinValue;
+            if (aRecent != bRecent)
+                return aRecent ? -1 : 1;
+
+            // Sort by connection status. Disconnected players will be last.
+            if (a.Connected != b.Connected)
+                return a.Connected ? -1 : 1;
+
+            // Sort connected players by whether they have joined the round, then by New Player status, then by Antag status
+            if (a.Connected && b.Connected)
+            {
+                var aNewPlayer = IsNewPlayer(a);
+                var bNewPlayer = IsNewPlayer(b);
+
+                //  Players who have joined the round will be listed before players in the lobby
+                if (a.ActiveThisRound != b.ActiveThisRound)
+                    return a.ActiveThisRound ? -1 : 1;
+
+                //  Within both the joined group and lobby group, new players will be grouped and listed first
+                if (aNewPlayer != bNewPlayer)
+                    return aNewPlayer ? -1 : 1;
+
+                //  Within all four previous groups, antagonists will be listed first.
+                if (a.Antag != b.Antag)
+                    return a.Antag ? -1 : 1;
+            }
+
+            // Sort disconnected players by participation in the round
+            if (!a.Connected && !b.Connected)
+            {
+                if (a.ActiveThisRound != b.ActiveThisRound)
+                    return a.ActiveThisRound ? -1 : 1;
+            }
+
+            // Finally, sort by the most recent message.
+            return bch.LastMessage.CompareTo(ach.LastMessage);
+        };
+
+        ViewVariables.OnPressed += _ =>
+        {
+            if (_currentPlayer is not null)
+                _console.ExecuteCommand($"vv \"{_currentPlayer.NetEntity}\"");
+        };
+
+        Follow.OnPressed += _ =>
+        {
+            if (_currentPlayer is not null)
+                _console.ExecuteCommand($"follow \"{_currentPlayer.NetEntity}\"");
+        };
+
+        Respawn.OnPressed += _ =>
+        {
+            if (_currentPlayer is not null)
+                _console.ExecuteCommand($"respawn \"{_currentPlayer.Username}\"");
+        };
+
+        PopOut.OnPressed += _ =>
+        {
+            uiController.PopOut();
+        };
+    }
+
+    public void OnCwoink(NetUserId channel)
+    {
+        ChannelSelector.PopulateList();
+    }
+
+
+    public void SelectChannel(NetUserId channel)
+    {
+        if (!Extensions.TryFirstOrDefault<PlayerInfo>(ChannelSelector.PlayerInfo, i => i.SessionId == channel, out var info))
+            return;
+
+        // clear filter if we're trying to select a channel for a player that isn't currently filtered
+        // i.e. through the message verb.
+        var data = new PlayerListData(info);
+        if (!Enumerable.Contains<ListData>(ChannelSelector.PlayerListContainer.Data, data))
```

Comment:
Name can be simplified
```suggestion
        if (!Enumerable.Contains(ChannelSelector.PlayerListContainer.Data, data))
```